lib/tests/test_type_hints.c

```diff
@@ -214,14 +214,31 @@ ParameterizedTestParameters(type_hints, test_double_cast)
                              sizeof(string_value_pairs) / sizeof(string_value_pairs[0]));
 }
 
+static void
+cr_assert_gdouble_eq(gdouble a, gdouble b)
+{
+  const gint is_a_inf = isinf((long double)a);
+  const gint is_b_inf = isinf((long double)a);
+
+  cr_assert_eq(is_a_inf, is_b_inf);
+  if (is_a_inf && is_b_inf)
+    {
+      cr_assert(TRUE);
+      return;
+    }
+
+  cr_assert(fabs(a-b) < G_MINDOUBLE);
```

Comment:
Me feels G_MINDOUBLE too small, but have no better idea. Nonetheless it is equivalent change this way. Later we might want to pass this as a parameter if we run into trouble due to precision.